lib/pdfkit/source.rb

```diff
@@ -46,7 +46,8 @@ def shell_safe_url
     end
 
     def url_needs_escaping?
-      URI::DEFAULT_PARSER.unescape(@source) == @source
+      URI::DEFAULT_PARSER.unescape(@source) == @source ||
+        URI::DEFAULT_PARSER.escape(URI::DEFAULT_PARSER.unescape(@source)) != @source
```

Comment:
the change on line 50 looks good, but the original condition on line 49 now looks redundant

Reply:
Thanks for the review @richardwan, I've fixed it.